app/scripts/components/app/edit-raw.js

```diff
@@ -123,20 +123,18 @@ const EditRaw = React.createClass({
                 maxLines={200}
               />
               <br />
-              <span className={'button button__animation--md button__arrow button__arrow--md button__animation button__arrow--white' + (updateStatus === 'inflight' ? ' button--disabled' : '')}>
+              <span onClick={this.submit} className={'button button__animation--md button__arrow button__arrow--md button__animation button__arrow--white' + (updateStatus === 'inflight' ? ' button--disabled' : '')}>
```

Comment:
Is the reason for moving the `onClick` handler because the `input` element doesn't take up the full size of the `span`, so that not all of the animated button was actually clickable?

Also, I know you didn't write this code, but can we test whether it would work to use `button` element instead? Or apply the classes to the `input` element directly so that the whole thing is clickable? Using `span` or generally anything other than `button`/`input` for buttons is bad for HTML semantics and really, really bad for accessibility

Reply:
@markdboyd    
`Is the reason for moving the onClick handler because the input element doesn't take up the full size of the span, so that not all of the animated button was actually clickable?`

Yes, the animation arrow element gets added to the span, which results in button size shenanigans on hover (and it's not clickable either) if you simply grow the input element.

Reply:
In anycase, you're entirely correct, I'll remove the span and move all of this into a button element.